llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -1782,31 +1951,32 @@ bool PreRARematStage::canIncreaseOccupancyOrReduceSpill() {
       if (!isTriviallyReMaterializable(DefMI))
         continue;
 
-      // We only support rematerializing virtual registers with one definition.
+      // We only support rematerializing virtual registers with one
+      // definition.
```

Comment:
Not directly related to this PR, but I think we also need to check either that the use operands of the remat candidate only have 1 def, or, preferably, when doing the remat we check that the reaching def of the uses is the same as the original def.